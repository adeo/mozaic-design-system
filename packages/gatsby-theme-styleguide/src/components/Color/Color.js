import styled, { css } from 'styled-components'
import React, { PureComponent } from 'react'

import Copy from '../Copy'

const Container = styled.div`
  border-radius: 4px;
  overflow: hidden;
  box-shadow: 1px 2px 2px 0 rgba(0, 0, 0, 0.1);
  border: solid 1px rgba(0, 0, 0, 0.05);
  padding: 0;
`
const Swatch = styled.div`
  width: 100%;
  padding-bottom: 8%;
  max-height: 150px;
  margin-bottom: 10px;

  ${({ color }) =>
    color &&
    css`
      background: ${color};
    `};
`

const Label = styled.div`
  display: inline-block;
  min-width: 35px;
  text-transform: uppercase;
  text-align: center;
  margin-right: 10px;
  padding-right: 10px;
  border-right: solid 1px #f5f5f5;
`

class Color extends PureComponent {
  render() {
    const { color, scss, ios, android, es6 } = this.props

    return (
      <Container>
        <Swatch color={color} />
        <Copy value={color} children={<Value label="val" value={color} />} />
        <Copy value={scss} children={<Value label="scss" value={scss} />} />
        <Copy value={ios} children={<Value label="ios" value={ios} />} />
        <Copy
          value={android}
          children={<Value label="andr" value={android} />}
        />
        <Copy value={es6} children={<Value label="es6" value={es6} />} />
      </Container>
    )
  }
}

const Value = ({ label, value }) => {
  return (
    <span>
      <Label>
        <span>{label}</span>
      </Label>
      <b>{value}</b>
    </span>
  )
}

export default Color
