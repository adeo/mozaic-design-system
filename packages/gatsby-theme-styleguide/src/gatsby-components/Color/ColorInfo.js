import styled from 'styled-components'
import React, { PureComponent } from 'react'

const Info = styled.div`
  width: 100%;
  padding: 3px 5px;
  font-size: 10px;
  color: #999;
  height: 2em;
  line-height: 2em;
  cursor: pointer;

  &:not(:last-child) {
    border-bottom: solid 1px #f5f5f5;
  }
`

const Copied = styled.div`
  display: inline-block;
  color: #f5f5f5;
  background: #999;
  border-radius: 3px;
  padding: 3px;
  line-height: 1;
  font-size: 9px;
  margin-left: 10px;
`

const Label = styled.div`
  display: inline-block;
  min-width: 35px;
  text-transform: uppercase;
  text-align: center;
  margin-right: 10px;
  padding-right: 10px;
  border-right: solid 1px #f5f5f5;
`

class Color extends PureComponent {
  constructor(props) {
    super(props)
    this.state = {
      copyCompatible: false,
      copied: false,
    }
  }

  componentDidMount() {
    this.isClipBoardAPIAvailable()
  }

  isClipBoardAPIAvailable = () => {
    if (navigator.permissions && navigator.permissions.query) {
      navigator.permissions.query({ name: 'clipboard-write' }).then(result => {
        if (result.state === 'granted' || result.state === 'prompt') {
          this.setState({ copyCompatible: true })
        }
      })
    }
  }

  copyToClipBoard = () => {
    if (this.state.copyCompatible) {
      navigator.clipboard.writeText(this.props.value).then(
        () => {
          this.setState({ copied: true }, () => {
            setTimeout(() => this.setState({ copied: false }), 1000)
          })
        },
        () => {
          console.error('error while copying to clipboard')
        }
      )
    }
  }

  render() {
    const { value, label } = this.props
    const { copied } = this.state

    return (
      <Info onClick={this.copyToClipBoard}>
        <Label>
          <span>{label} : </span>
        </Label>
        <b>{value}</b>
        {copied && <Copied>copied</Copied>}
      </Info>
    )
  }
}

export default Color
