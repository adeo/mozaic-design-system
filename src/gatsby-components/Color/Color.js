import styled, { css } from 'styled-components'
import React, { PureComponent } from 'react'

import Info from './ColorInfo'

const Container = styled.div`
  border: solid 1px #f5f5f5;
  border-radius: 3px;
  padding: 5px;
  margin: 20px 0;
`

const Swatch = styled.div`
  width: 100%;
  height: 50px;
  margin-bottom: 10px;

  ${({ color }) =>
    color &&
    css`
      background: ${color};
    `};
`

class Color extends PureComponent {
  render() {
    const { color, name } = this.props

    return (
      <Container>
        <Swatch color={color} />
        <Info label="value" value={color} />
        <Info label="name" value={name} />
      </Container>
    )
  }
}

export default Color
