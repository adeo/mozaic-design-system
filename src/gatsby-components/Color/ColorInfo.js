import styled from 'styled-components'
import React, { PureComponent } from 'react'

const Info = styled.div`
  width: 100%;
  font-size: 14px;
  height: 2em;
  line-height: 2em;
  cursor: pointer;

  &:not(:last-child) {
    border-bottom: solid 1px #f5f5f5;
  }
`

const Copied = styled.div`
  display: inline-block;
  color: #f5f5f5;
  background: #999;
  border-radius: 3px;
  padding: 3px;
  line-height: 1;
  font-size: 14px;
  margin-left: 10px;
`

class Color extends PureComponent {
  constructor(props) {
    super(props)
    this.state = {
      copyCompatible: false,
      copied: false,
    }
  }

  componentDidMount() {
    this.isClipBoardAPIAvailable()
  }

  isClipBoardAPIAvailable = () => {
    if (navigator.permissions && navigator.permissions.query) {
      navigator.permissions.query({ name: 'clipboard-write' }).then(result => {
        if (result.state === 'granted' || result.state === 'prompt') {
          this.setState({ copyCompatible: true })
        }
      })
    }
  }

  copyToClipBoard = () => {
    if (this.state.copyCompatible) {
      navigator.clipboard.writeText(this.props.value).then(
        () => {
          this.setState({ copied: true }, () => {
            setTimeout(() => this.setState({ copied: false }), 1000)
          })
        },
        () => {
          console.error('error while copying to clipboard')
        }
      )
    }
  }

  render() {
    const { value, label } = this.props
    const { copied } = this.state

    return (
      <Info onClick={this.copyToClipBoard}>
        {label}: <b>{value}</b>
        {copied && <Copied>copied</Copied>}
      </Info>
    )
  }
}

export default Color
