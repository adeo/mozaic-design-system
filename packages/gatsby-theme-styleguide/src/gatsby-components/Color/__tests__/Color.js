import React from 'react';
import Color from '../Color';
import renderer from 'react-test-renderer';

test('Test color component', () => {
  global.navigator = jest.fn();

  const component = renderer.create(
    <Color color="blue" />,
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});