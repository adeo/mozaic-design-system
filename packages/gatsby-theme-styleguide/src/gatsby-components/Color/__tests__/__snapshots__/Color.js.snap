// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test color component 1`] = `
<div
  className="sc-htpNat gIKUYi"
>
  <div
    className="sc-bxivhb llJfDc"
    color="blue"
  />
  <div
    className="sc-bdVaJa iYZTXX"
    onClick={[Function]}
  >
    <span>
      <div
        className="sc-ifAKCX iHSdmX"
      >
        <span>
          val
        </span>
      </div>
      <b>
        blue
      </b>
    </span>
  </div>
  <div
    className="sc-bdVaJa iYZTXX"
    onClick={[Function]}
  >
    <span>
      <div
        className="sc-ifAKCX iHSdmX"
      >
        <span>
          scss
        </span>
      </div>
      <b />
    </span>
  </div>
  <div
    className="sc-bdVaJa iYZTXX"
    onClick={[Function]}
  >
    <span>
      <div
        className="sc-ifAKCX iHSdmX"
      >
        <span>
          ios
        </span>
      </div>
      <b />
    </span>
  </div>
  <div
    className="sc-bdVaJa iYZTXX"
    onClick={[Function]}
  >
    <span>
      <div
        className="sc-ifAKCX iHSdmX"
      >
        <span>
          andr
        </span>
      </div>
      <b />
    </span>
  </div>
  <div
    className="sc-bdVaJa iYZTXX"
    onClick={[Function]}
  >
    <span>
      <div
        className="sc-ifAKCX iHSdmX"
      >
        <span>
          es6
        </span>
      </div>
      <b />
    </span>
  </div>
</div>
`;
